ci/unit/test_post_always.sh

```diff
@@ -17,12 +17,7 @@ DAOS_BASE="${SL_PREFIX%/install*}"
 NODE="${NODELIST%%,*}"
 mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
-
-if [ "$WITH_VALGRIND" == "memcheck" ]; then
-    rm -rf run_test_memcheck.sh
-else
-    rm -rf run_test.sh vm_test
-fi
+rm -rf run_test.sh vm_test run_test_memcheck.sh
```

Comment:
if they use the same workspaces, you may have to only remove the one we just ran